app/tls_crypt_v2.ml

```diff
@@ -133,6 +158,65 @@ let load_tls_crypt_v2_server_key filename =
   let* key = load_pem ~name:"OpenVPN tls-crypt-v2 server key" filename in
   Tls_crypt_v2_key.of_cstruct (Cstruct.of_string key)
 
+let line oc str =
+  output_string oc str;
+  output_byte oc (Char.code '\n')
+
+let generate_tls_crypt_v2_server_key ?g () = Tls_crypt_v2_key.generate ?g ()
+
+let save_tls_crypt_v2_server_key server_key filename =
+  let oc = open_out filename in
+  line oc "-----BEGIN OpenVPN tls-crypt-v2 server key-----";
+  let b64 = Tls_crypt_v2_key.to_base64 server_key in
+  let lines = String.split_at 65 b64 in
+  List.iter (line oc) lines;
+  line oc "-----END OpenVPN tls-crypt-v2 server key-----";
+  close_out oc
+
+let generate_tls_crypt_v2_client_key ?g metadata =
+  let metadata = match metadata with
+    | None -> Metadata.now ()
+    | Some metadata -> metadata in
+  let a = Tls_crypt_v2_key.generate ?g () in
+  let b = Tls_crypt_v2_key.generate ?g () in
+  (a, b, metadata)
+
+let tls_crypt_v2_wrap_client server_key (a, b, metadata) =
+  let a = Tls_crypt_v2_key.to_cstruct a in
+  let b = Tls_crypt_v2_key.to_cstruct b in
+  let cs = Cstruct.concat [ a; b ] in
+  let metadata = Metadata.to_cstruct metadata in
+  let net_len = (128 * 2) + (Cstruct.length metadata) + _TLS_CRYPT_V2_TAG_SIZE + 2 in
```

Comment:
Not necessarily clearer:
```suggestion
  let net_len = Cstruct.length a + Cstruct.length b + Cstruct.length metadata + Mirage_crypto.Hash.SHA256.digest_size + 2 in
```

Reply:
Or since we bind `cs := a||b`
```suggestion
  let net_len = Cstruct.length cs + Cstruct.length metadata + Mirage_crypto.Hash.SHA256.digest_size + 2 in
```